File: lib/startup/restoreOverwrittenFilesWithOriginals.js
/*
 * Copyright (c) 2014-2020 Bjoern Kimminich.
 * SPDX-License-Identifier: MIT
 */

const path = require('path')
const fs = require('fs')
const logger = require('../logger')
const { promisify } = require('util')
const glob = promisify(require('glob'))
const copyFile = promisify(fs.copyFile)
const access = promisify(fs.access)

const exists = (path) => access(path).then(() => true).catch(() => false)

const restoreOverwrittenFilesWithOriginals = async () => {
  await copyFile(path.resolve(__dirname, '../../data/static/legal.md'), path.resolve(__dirname, '../../ftp/legal.md'))

  if (await exists(fs.existsSync(path.resolve(__dirname, '../../frontend/dist')))) {
    copyFile(path.resolve(__dirname, '../../data/static/JuiceShopJingle.vtt'), path.resolve(__dirname, '../../frontend/dist/frontend/assets/public/videos/JuiceShopJingle.vtt'))
  }

  try {
    const files = await glob(path.join(__dirname, '../../data/static/i18n/*.json'))
    await Promise.all(
      files.map((filename) => copyFile(filename, path.resolve(__dirname, '../../i18n/', filename.substring(filename.lastIndexOf('/') + 1))))
    )
  } catch (err) {
    logger.warn('Error listing JSON files in /data/static/i18n folder: ' + err.message)
  }
}

module.exports = restoreOverwrittenFilesWithOriginals

File: lib/startup/restore_overwritten_files_with_originals.py
import path
import utils
import logger
from shutil import copyfile
from glob import glob

def exists(path):
    try:
        with open(path):
            return True
    except IOError:
        return False

def restore_overwritten_files_with_originals():
    copyfile(path.resolve('data/static/legal.md'), path.resolve('ftp/legal.md'))

    if exists(path.resolve('frontend/dist')):
        copyfile(
            path.resolve('data/static/owasp_promo.vtt'),
            path.resolve('frontend/dist/frontend/assets/public/videos/owasp_promo.vtt')
        )

    try:
        files = glob(path.resolve('data/static/i18n/*.json'))
        for filename in files:
            copyfile(filename, path.resolve('i18n/', filename[filename.rfind('/') + 1:]))
    except Exception as err:
        logger.warn('Error listing JSON files in /data/static/i18n folder: ' + utils.get_error_message(err))
